src/main/java/org/molgenis/utils/HgncFile.java

```diff
@@ -0,0 +1,117 @@
+package org.molgenis.utils;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileReader;
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class HgncFile {
+
+  private static final String PREVIOUS = "previous_symbol";
+  private static final String ALIAS = "alias_symbol";
+  private static final String SYMBOL = "approved_symbol";
+  private static final String SEPARATOR = "\t";
+  private static final String EMPTY = "";
+
+  private HashMap<String, Integer> headerPositions;
+  private HashMap<String, String> alternativeGeneNames;
+  private HashMap<String, HashMap<String, String>> genes;
+
+  public HgncFile(String geneFilePath) throws IOException {
+    this.setAlternativeGeneNames(new HashMap<>());
+    this.setHeaderPositions(new HashMap<>());
+    this.setGenes(new HashMap<>());
+    this.getGeneData(geneFilePath);
+  }
+
+  public Map<String, HashMap<String, String>> getGenes() {
+    return this.genes;
+  }
+
+  private void setGenes(HashMap<String, HashMap<String, String>> genes) {
+    this.genes = genes;
+  }
+
+  private void addToGenes(HashMap<String, String> geneInfo) {
+    this.genes.put(geneInfo.get(SYMBOL), geneInfo);
+  }
+
+  private Map<String, Integer> getHeaderPositions() {
+    return this.headerPositions;
+  }
+
+  private void setHeaderPositions(HashMap<String, Integer> headerPositions) {
+    this.headerPositions = headerPositions;
+  }
+
+  private void addToHeaderPositions(String header, Integer position) {
+    this.headerPositions.put(header, position);
+  }
+
+  public Map<String, String> getAlternativeGeneNames() {
+    return this.alternativeGeneNames;
+  }
+
+  private void addToAlternativeGeneName(String alternative, String original) {
+    this.alternativeGeneNames.put(alternative.toLowerCase(), original);
+  }
+
+  private void setAlternativeGeneNames(HashMap<String, String> alternativeGeneNames) {
+    this.alternativeGeneNames = alternativeGeneNames;
+  }
+
+
+  private void setHeaderPositions(String header) {
+    String[] columns = header.toLowerCase().replace(" ", "_").split(SEPARATOR);
+    int position = 0;
+    for (String key : columns) {
+      this.addToHeaderPositions(key, position);
+      position++;
+    }
+  }
+
+  private void getGeneData(String geneFilePath) throws IOException {
+    File geneFile = new File(geneFilePath);
+    try (BufferedReader br = new BufferedReader(new FileReader(geneFile))) {
```

Comment:
This will read the file using the platform's default charset.
```suggestion
    try (BufferedReader br = Files.newBufferedReader(geneFile, StandardCharsets.UTF_8)) {
```